gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+        "Professional_Summary\n",
+        "\n",
+        "--------------------------------------------------\n",
+        "ARRANGEMENT, TRUNCATION, GENERATION\n",
+        "--------------------------------------------------\n",
+        "Coursework\n",
+        "Skills\n",
+        "Certifications\n",
+        "\n",
+        "\"\"\"\n",
+        "\n",
+        "PROMPT_EXTR = \"\"\"\n",
+        "You are a seasoned career advising professional expert in crafting resumes and cover letters.\n",
+        "\n",
+        "Consider the below details that have been extracted from a resume\n",
+        "%s\n",
+        "\n",
+        "INSTRUCTIONS:\n",
+        "- Maintain truthfulness and objectivity.\n",
+        "- Prefer active voice over passive voice.\n",
+        "- The output should have the following valid JSON format\n",
+        "{\n",
+        "\"Personal_Information\": {\"Name\": \"\", \"Address\": \"\", \"Mobile_No\": \"\", \"Email\": \"\"},\n",
+        "\"Education\": \"\",\n",
+        "\"Experience\": \"\",\n",
+        "\"Languages\": \"\",\n",
+        "\"Extracurricular_Activities\": \"\",\n",
+        "\"References\": \"\",\n",
+        "}\n",
+        "- Output the Education, Experience, Languages, Extracurricular_Activities and References\n",
+        "sections as a simple numbered list.\n",
+        "\"\"\"\n",
+        "\n",
+        "PROMPT_GEN = \"\"\"\n",
+        "You are a seasoned career advising professional expert in crafting resumes and cover letters.\n",
+        "\n",
+        "Consider the below details that have been extracted from a resume\n",
+        "%s\n",
+        "\n",
+        "Use the above details and craft a resume to satisfy the job description provided below\n",
+        "%s\n",
+        "\n",
+        "INSTRUCTIONS:\n",
+        "- Maintain truthfulness and objectivity.\n",
+        "- Prefer active voice over passive voice.\n",
+        "- Generate a concise professional summary in 200 words based on the job description and the\n",
+        "extracted resume.\n",
+        "- Return the professional summary as a paragraph without any headers.\n",
+        "\"\"\"\n",
+        "\n",
+        "PROMPT_ARR = \"\"\"\n",
+        "You are a seasoned career advising professional expert in crafting resumes and cover letters.\n",
+        "\n",
+        "Consider the below details that have been extracted from a resume\n",
+        "%s\n",
+        "\n",
+        "Use the above details and craft a resume to satisfy the job description provided below\n",
+        "%s\n",
+        "\n",
+        "INSTRUCTIONS:\n",
+        "- Maintain truthfulness and objectivity.\n",
+        "- Proofread and correct spelling and grammatical errors.\n",
+        "- Prefer passive voice over active voice.\n",
+        "- The output should have the following valid JSON format\n",
+        "{\n",
+        "\"Coursework\": \"\",\n",
+        "\"Skills\": \"\",\n",
+        "\"Certifications\": \"\",\n",
+        "}\n",
```

Comment:
Same as above, use Controlled Generation to enforce a specific schema.

Reply:
Replaced with GenerationConfig class and added controlled generation across the code